src/components/views/elements/MemberEventListSummary.js

```diff
@@ -144,87 +144,105 @@ module.exports = React.createClass({
         );
     },
 
-    render: function() {
-        let summary = null;
+    shouldComponentUpdate: function(nextProps, nextState) {
+        return (
+            nextProps.events.length !== this.props.events.length ||
```

Comment:
Sorry, I didn't mean children, I meant inner components. I wasn't referring to the `EventTile`s passed through, I was referring to the `ScrollPanel` updating (as the parent) and  `MemberEventListSummary` re-rendering.

> yup... but I'm not sure if that's a worthwhile saving, having fixed render.

It's still quite bad for large number of membership events, say 5000, which will be broken several times over when we do initial membership syncing on freenode. It's bad enough waiting for the view to re-render a single summary every time there's a new membership event, let alone re-rendering every summary every time a message is put into the `ScrollPanel` or a read receipt changes position.